app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,35 @@
+class MarketsController < ApplicationController
+  before_action :set_nav
+  before_action :set_market, only: [:show, :edit, :update]
+
+  def index
+    @markets = Market.all
+  end
+
+  def new
+    @market = Market.new
+  end
+
+  def create
+    market = Market.create(market_params)
+    redirect_to market
+  end
```

Comment:
You can do some nifty stuff with `render` in these cases if you keep your variables consistent between `#new` and `#create` (and similarly between `#edit` and `#update`).